server/pbench/bin/pbench-server-prep-shim-001

```diff
@@ -0,0 +1,220 @@
+#! /bin/bash
+
+# set -x
+
+# This shim is used to prepare the tarballs that a version 001 client
+# submits for further processing. It copies the tarballs and their MD5
+# sums (plus an optional prefix file) to the archive (after checking)
+# and sets the state links, so that the dispatch script will pick them
+# up and get the ball rolling. IOW, it does impedance matching between
+# version 001 clients and the server scripts.
+
+version=001
+
+# There is an enormous amount of code duplication between this script
+# and the version 002 script. TBD: refactor (but maybe wait and do it
+# for the inotify versions - the crontab version has a very limited life
+# span at this point).
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+. $dir/pbench-base.sh
+###########################################################################
+
+archive=$(getconf.py archive-dir pbench-server)
+quarantine=$(getconf.py quarantine-dir pbench-server)
+duplicates=$(getconf.py duplicates-dir pbench-server)
+
+errlog=$LOGSDIR/$PROG/$PROG.error
+if [[ "$archive" == "" ]] ;then
+    echo "Failed: \"getconf.py archive-dir pbench-server" >> $errlog
+    exit 1
+fi
+
+if [[ "$quarantine" == "" ]] ;then
+    echo "Failed: \"getconf.py quarantine-dir pbench-server" $errlog
+    exit 1
+fi
+
+if [[ "$duplicates" == "" ]] ;then
+    echo "Failed: \"getconf.py duplicates-dir pbench-server" $errlog
+    exit 1
+fi
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+tmp=$TMP/$PROG.$$
+
+function quarantine () {
+    dest=$1
+    shift
+    files="$@"
+
+    mkdir -p $dest || return $?
+    mv $files $dest || return $?
+}
+
+trap 'rm -rf $tmp' EXIT INT QUIT
+
+mkdir -p $tmp
+list=$tmp/list.tb
+status=$tmp/status
+> $status
+# we are explicitly handling version-001 data
+results_dir=$(getconf.py receive-dir-${version} pbench-server)
```

Comment:
This appears to be the only place we reference `${version}`.  Can we consider getting rid of that variable and using `receive-dir-001` instead?  We already reference 001 in other comments, so the variable does not appear to be that useful.

Reply:
OK